File: 1726-tuple-with-same-product/main.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        
        freq = defaultdict(int)
        n = len(nums)
        for i in range(n - 1):
            for j in range(i + 1, n):
                freq[nums[i] * nums[j]] += 1

        count = 0
        for product, frequency in freq.items():
            count += 8 * (frequency * (frequency-1)) / 2

        return int(count)

File: 1726-tuple-with-same-product/main.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        
        size_t n = nums.size();
        unordered_map<int, int> freq;
        for (size_t i = 0; i < n - 1; i++) {
            for (size_t j = i + 1; j < n; j++) {
                freq[nums[i] * nums[j]]++;
            }
        }

        int count = 0;
        for (auto [product, frequency]: freq) {
            count += 8 * (frequency * (frequency - 1)) / 2;
        }

        return count;
    }
};
